File: Python/Base_io.py
class BaseIO:
    def __init__(self, last_device="", last_output=""):
        self._last_device = last_device
        self._last_output = last_output

    def get_last_device(self):
        return self._last_device

    def get_last_output(self):
        return self._last_output

    def set_last_device(self, last_device):
        self._last_device = last_device

    def set_last_output(self, last_output):
        self._last_output = last_output

    def info(self):
        return f"BaseIO: {self._last_device} {self._last_output}"

File: java/BaseIO.java
public class BaseIO {
    private String lastDevice;
    private String lastOutput;

    public BaseIO() {
        this.lastDevice = "";
        this.lastOutput = "";
    }

    public BaseIO(String lastDevice, String lastOutput) {
        this.lastDevice = lastDevice;
        this.lastOutput = lastOutput;
    }

    public String getLastDevice() {
        return lastDevice;
    }

    public void setLastDevice(String lastDevice) {
        this.lastDevice = lastDevice;
    }

    public String getLastOutput() {
        return lastOutput;
    }

    public void setLastOutput(String lastOutput) {
        this.lastOutput = lastOutput;
    }

    public String info() {
        return "BaseIO: " + lastDevice + " " + lastOutput;
    }
}